### DISCUSSION TITLE
Allow customizing pydantic's `ValidationError.errors()` arguments for errors wrapped up in `RequestValidationError`.

### DISCUSSION THREAD

**Discussion by benwah** [fastapi/fastapi#10934] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# Maybe something like this:

app = FastAPI(
    ...,
    "validation_error_arguments": {
        "include_url": False,
        "include_context": True,
        "include_input": True,
    },
)

# Or alternatively maybe we pass the original `ValidationError` in the `RequestValidation` context, then we could do something like this;


@app.exception_handler(RequestValidationError)
async def handle_request_validation_error(
    request: Request, exc: RequestValidationError
) -> JSONResponse:
    error: ErrorResponse = {
        "code": ErrorCodes.INVALID_PAYLOAD,
        "message": "Invalid request.",
        "details": None,
        "path": jsonable_encoder(exc.errors(include_urls=False)),
    }
    return JSONResponse(error, status_code=HTTPStatus.UNPROCESSABLE_ENTITY)
```

### Description

When a `RequestValidationError` is raised, there's no way to change the call to pydantic's `ValidationError.errors()`, which support a [few customization options](https://docs.pydantic.dev/dev/api/pydantic_core/#pydantic_core.ValidationError.errors).

This results in the following kind of error messages in the JSON payload when the request doesn't validate:

```json
                [
                    {
                        "input": {},
                        "loc": ["props", "value", "fooId"],
                        "msg": "Field required",
                        "type": "missing",
                        "url": "https://errors.pydantic.dev/2.4/v/missing",  # <-- I want to remove this.
                    }
                ]
```

This means that the only way to remove, for instance, the "url" field in the JSON response, is to remove the `"url"` item in each dictionary entry returned by `RequestValidationError.errors()`.

It would be nice to be able to customize this behaviour globally.

### Operating System

macOS

### FastAPI Version

0.108.0

### Pydantic Version

2.5.3

### Python Version

3.12.1

**Comment by k4nar** [comment]
The only workaround I’ve found is the following, but it’s clearly less elegant than using `include_urls` or `include_context` :confused: :
```python
errors = jsonable_encoder(exc.errors(), exclude={"__all__": {"ctx", "url"}, "ctx": True, "url": True})
```
Note that `ctx` can contain stuff which aren’t JSON serializable, such as `ValueError` instances.

**Reply by k4nar** [reply]
Actually, this could be fixed with #11542 .

**Reply by bertomaniac** [reply]
I tried that PR, but it didn't work for my case.

I documented my fix in this discussion: https://github.com/fastapi/fastapi/discussions/11923

And, @k4nar, you're right. `ctx` can contain stuff which isn't json serializable. I provided a test to show exactly that in the linked discussion.

It would be nice if we could provide a global option to drop the `url`, `ctx`, `input` generated from pydantic v2 errors.

**Comment by YuriiMotov** [comment, marked as answer]
`url` is [excluded from validation errors response](https://fastapi.tiangolo.com/release-notes/#01102) since FastAPI 0.110.2 🎉

To exclude other fields you can use the following approach:

```python
from typing import Any
from fastapi import FastAPI, Request, status
from fastapi.encoders import jsonable_encoder
from fastapi.exceptions import RequestValidationError
from fastapi.responses import JSONResponse

app = FastAPI()


def exclude_input(err: dict[str, Any]) -> dict[str, Any]:
    return {
        field_name: field_value
        for field_name, field_value in err.items()
        if field_name != "input"
    }



@app.exception_handler(RequestValidationError)
async def validation_exception_handler(request: Request, exc: RequestValidationError):
    return JSONResponse(
        status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
        content=jsonable_encoder({"detail": [exclude_input(err) for err in exc.errors()]}),
    )

@app.post("/items/{item_id}")
async def create_item(item_id: int):
    return item_id


```

**Note:** don't forget to update response model to make it compatible with the data you return!
